enhancements/olm/catalogd-query-endpoint.md

```diff
@@ -0,0 +1,585 @@
+            application/jsonl:
+              schema:
+                type: string
+          description: JSONL-formatted schemas stream, or empty if no match
+      operationId: queryCatalog
+      description: Retrieve catalog content corresponding to a catalog content query by name, schema, or package name
+    head:
+      responses:
+        "200":
+          headers:
+            Last-Modified:
+              schema:
+                format: date-time
+                type: string
+          description: Response headers as if the request were performed
+    parameters:
+      - name: baseURL
+        description: The base API URL from the object's status.urls.base
+        schema:
+          type: string
+        in: path
+        required: true
+      - examples:
+          all:
+            value: 'Accept: */*'
+        name: Accept
+        description: Communicates the response encoding(s) this client handles
+        schema:
+          type: string
+        in: header
+        required: false
+  /{baseURL}/api/v1/all:
+    get:
+      responses:
+        "200":
+          headers:
+            Last-Modified:
+              schema:
+                format: date-time
+                type: string
+          content:
+            application/jsonl:
+              schema:
+                type: string
+          description: JSONL-formatted catalog FBC
+        "404":
+          content:
+            text/plain:
+              schema:
+                type: string
+          description: ""
+      operationId: getCatalogJSONL
+      description: Retrieve all catalog content
+    parameters:
+      - name: baseURL
+        description: The base API URL from the object's status.urls.base
+        schema:
+          type: string
+        in: path
+        required: true
+```
+<!--
+What are some important details that didn't come across above in the
+**Proposal**? Go in to as much detail as necessary here. This might be
+a good place to talk about core concepts and how they relate. While it is useful
+to go into the details of the code changes required, it is not necessary to show
+how the code will be rewritten in the enhancement.
+-->
+
+### Caching Considerations
+
+This proposal indends to provide RFC7234 caching compliance through support for `Last-Modified` and `If-Modified-Since` directives. Clients can use these headers to avoid re-downloading unchanged data. 
+The server will respond with 304 Not Changed if the catalog metadata is unchanged.  
+                  
+Clients are also encouraged to implement local caching for frequently queried metadata.
+
+
+### Risks and Mitigations
+
+Depending on the implementation of this proposal, exercise of this endpoint could function as an I/O multiplier which could starve other clients which are dependent on the same API. 
+The first iteration of this endpoint will not attempt to mitigate denial-of-service attempts via rate-limiting or other resource consumption constraints.
+
```

Comment:
Might be useful to point out that access to catalog metadata via both `all` and this new endpoint is unrestricted and does not require authentication or authorization, and that we do not (yet at least) have use cases that would require restricted access based on user identity.